From adf28739565c406fdb957755a2a7324a55f94724 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 03 Sep 2021 19:23:34 +0300
Subject: [PATCH] [SW-0] arc: simpify copy_preboot() logic

Simplify the logic of the condition and as a benefit get lower
indentation of the code inside the condition.
Additional benefit is a better readability of this code.

Note: clearly it is the same code that should be shared for all ARC
architectures, but this commit should not deal with that kind of
consolidation.

Change-Id: Ie8ed4ee9ba3b588011d4086a3106be67a193aa1b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 8573e73..42d4f73 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -230,17 +230,12 @@
 	u32 config_fuse;
 	int ret;
 
-	if (ppboot_bringup_preboot(ARCBOOT_PRIMARY) != ARCBOOT_BRINGUP_OK) {
-		if (ppboot_bringup_preboot(ARCBOOT_SECONDARY) !=
-			ARCBOOT_BRINGUP_OK) {
-			/* if debug I/F is enabled then fall back on I2C_S */
-			ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET,
-						&config_fuse);
-			if (!ret && config_fuse !=
-					EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) {
-				security_i2c_s_enable();
-			}
-		}
+	if (ppboot_bringup_preboot(ARCBOOT_PRIMARY) != ARCBOOT_BRINGUP_OK &&
+	    ppboot_bringup_preboot(ARCBOOT_SECONDARY) != ARCBOOT_BRINGUP_OK) {
+		/* if debug I/F is enabled then fall back on I2C_S */
+		ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_fuse);
+		if (!ret && config_fuse != EFUSE_SYSCFG_DISABLE_TWI_IF_MASK)
+			security_i2c_s_enable();
 	}
 
 	return PREBOOT_TEXT_BASE;
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 06b05e8..1c1c171 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -230,17 +230,12 @@
 	u32 config_fuse;
 	int ret;
 
-	if (ppboot_bringup_preboot(ARCBOOT_PRIMARY) != ARCBOOT_BRINGUP_OK) {
-		if (ppboot_bringup_preboot(ARCBOOT_SECONDARY) !=
-			ARCBOOT_BRINGUP_OK) {
-			/* if debug I/F is enabled then fall back on I2C_S */
-			ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET,
-						&config_fuse);
-			if (!ret && config_fuse !=
-					EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) {
-				security_i2c_s_enable();
-			}
-		}
+	if (ppboot_bringup_preboot(ARCBOOT_PRIMARY) != ARCBOOT_BRINGUP_OK &&
+	    ppboot_bringup_preboot(ARCBOOT_SECONDARY) != ARCBOOT_BRINGUP_OK) {
+		/* if debug I/F is enabled then fall back on I2C_S */
+		ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_fuse);
+		if (!ret && config_fuse != EFUSE_SYSCFG_DISABLE_TWI_IF_MASK)
+			security_i2c_s_enable();
 	}
 
 	return PREBOOT_TEXT_BASE;
